txn.go

```diff
@@ -192,6 +193,45 @@ func (txn *Txn) Union(columns ...string) *Txn {
 	return txn
 }
 
+// WithUnion computes a union between all given indexes, and then
+// applies the result to the txn index.
+func (txn *Txn) WithUnion(columns ...string) *Txn {
+	if !txn.setup || len(columns) == 1 {
+		return txn.Union(columns...)
+	}
+
+	// allocate slice of column pointers
+	cols := make([]*column, 0)
+	for _, columnName := range columns {
+		if idx, ok := txn.columnAt(columnName); ok {
+			cols = append(cols, idx)
+		}
+	}
+
+	// allocate temp bitmap for calcs
+	tmpMap := make(bitmap.Bitmap, 1)
```

Comment:
Since this is iterating by chunk, we'd need a bitmap of size 256 (16,384/64), unless I'm missing something.

Reply:
Your bitmap package automatically grows the temp map -  https://github.com/kelindar/bitmap/blob/2bbf5487c3a81312758c6b0e6d51caf56dff1b14/bitmap_amd64.go#L67

However, it's more beneficial for use cases above one full chunk to pre-allocate, so setting the initial size to 256 is better imo. It also helped me notice that I wasn't completely zeroing-out the temp map.